guard/src/commands/github_pull.rs

```diff
@@ -0,0 +1,110 @@
+use config::Config;
+use semver::{BuildMetadata, Prerelease, Version, VersionReq};
+
+/// This is a class for getting file from GitHub
+pub struct GitHubSource {
+    pub user: String,
+    pub repo: String,
+    pub file_name: String,
+    pub access_token: String,
+    pub version_needed: String
+}
+
+
+/// Constructor
+impl GitHubSource {
+    pub fn new(user: String, repo: String, file_name: String, access_token:String, version_needed: String) -> Self {
+        GitHubSource {
+            user,
+            repo,
+            file_name,
+            access_token,
+            version_needed
+        }
+    }
+
+    /// Function to print detail of the instance
+    pub fn to_string(&self) -> String {
+        format!("GitHubSource user({}) repo ({}) file_name({}) access_token({}) version_needed({})",
+                &self.user, &self.repo, &self.file_name, &self.access_token, &self.version_needed)
+    }
+
+    /// Function to read config from TOML
+    fn read_config(&self) ->GitHubSource {
```

Comment:
this method must be out in a separate struct which will become part of `utils` crate, since we are reading a config and configs could be generic and not specific to Github source.

What we should keep here instead is a constructor that accepts all parameters post sanity check. (looks like you already have  one on line 16.